spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala

```diff
@@ -1466,6 +1466,15 @@ trait DeltaSQLConfBase {
       .booleanConf
       .createWithDefault(true)
 
+  val DELTA_STREAMING_SINK_ALLOW_TYPE_CHANGES =
```

Comment:
This name confused me a little bit. "Type changes" to me suggest that the types in the table will be changed by streaming sink, but IIUC then this merely allows the types of the written dataframe to differ from the types in the Delta table. Maybe call this something like "add implicit casts"?